deep_regression.py

```diff
@@ -0,0 +1,251 @@
+    global_step = get_or_create_global_step()
+
+    # Define your exponentially decaying learning rate
+    lr = tf.train.exponential_decay(learning_rate=INITIAL_LEARNING_RATE,
+                                    global_step=global_step,
+                                    decay_steps=decay_steps,
+                                    decay_rate=LEARNING_RATE_DECAY_FACTOR,
+                                    staircase=True)
+
+# ********************** LOSS && OPTIMIZE *************************************
+
+    mean_squared_error_loss = tf.reduce_sum(
+        tf.square(tf.transpose(pred) - y),
+        name="loss") / (2 * total_observations)
+
+    regularization_loss = tf.add_n(tf.losses.get_regularization_losses())
+
+    total_loss = mean_squared_error_loss + regularization_loss
+    # total_loss = tf.losses.get_total_loss(add_regularization_losses=True)
+
+#                   ************************************
+
+    optimizer = tf.train.AdamOptimizer(learning_rate=lr).\
+        minimize(total_loss, global_step=global_step)
+
+# ******************************************************************************
+    # Launch the graph
+    with tf.Session() as sess:
+
+        sess.run(tf.global_variables_initializer())
+
+        # Create a saver object which will save all the variables
+        saver = tf.train.Saver()
+
+        for epoch in range(TRAINING_EPOCHS + 1):
+            avg_cost = 0.0
+            total_batch = int(total_observations / BATCH_SIZE)
+            for i in range(total_batch - 1):    # Loop over all batches
+                batch_x = x_train[i * BATCH_SIZE:(i + 1) * BATCH_SIZE]
+                batch_y = y_train[i * BATCH_SIZE:(i + 1) * BATCH_SIZE]
+                _, batch_cost, batch_pred = sess.run([optimizer,
+                                                      total_loss,
+                                                      pred],
+                                                     feed_dict={x: batch_x,
+                                                                y: batch_y})
+                # Run optimization (backprop) and cost op (to get loss value)
+                avg_cost += batch_cost / total_batch   # Compute average loss
+
+            if epoch % DISPLAY_STEP == 0:         # Display logs per epoch step
+                print("Epoch : ", "%05d" % (epoch + 1),
+                      " cost = ", " {:.9f} ".format(avg_cost))
+                for i in xrange(3):  # Comparing orig, predicted values
+                    print("label value:", batch_y[i], "predicted value:",
+                          batch_pred[i])
+                print("------------------------------------------------------")
+        print('Global_Step final value : ', sess.run(global_step))
+        # Saving the Graph and Variables in Checkpoint files
+        saver.save(sess, ckpt_dir + 'deep_regression_trained_model',
+                   global_step=global_step)
+        pred_out = sess.run(pred, feed_dict={x: x_train})
```

Comment:
Did you mean to feed in the test set this time? Because you have already evaluated pred on train set batch by batch.

Reply:
I already evaluated the train prediction batch by batch. But I need whole train dataset prediction values for visualizing. However, rather than appending those batch by batch values into another array, evaluating sess.run() seemed comfortable. Is this a bad style ?